astropy/io/tests/test_registry.py

```diff
@@ -410,3 +402,23 @@ class MyTable(Table):
         mt = MyTable([[1], [2]], names=['a', 'b'])
         mt.write(buffer, format='ascii')
         assert buffer.getvalue() == os.linesep.join(['a b', '1 2', ''])
+
+    def test_read_table_subclass_with_columns_attributes(self, tmpdir):
+        """Regression test for https://github.com/astropy/astropy/issues/7181
+        """
+
+        class MTable(Table):
+            pass
+
+        mt = MTable([[1, 2.5]], names=['a'])
+        mt['a'].unit = u.m
+        mt['a'].format = '.4f'
+        mt['a'].description = 'hello'
+
+        testfile = str(tmpdir.join('junk.fits'))
+        mt.write(testfile, overwrite=True)
+
+        t = MTable.read(testfile)
+        assert np.all(mt == t)
```

Comment:
You should check that the `mt['a']` attributes are the same.

Turns out that `mt == t` does not check column attributes, so apart from the fact that the read operation does not fail outright, you don't know that the unit survived serialization.